### PR TITLE
chore(storybook): react boilerplate

### LINKED ISSUE #263: Carbon Labs: Add react component boilerplate example
Blocked by #273

### PR DESCRIPTION
**Pull request by Gururajj77**
Closes #263 

adds react boilerplate example

#### Changelog

**New**
- `packages/react` folder
- an example component package built with `@carbon/react`
- snapshot and unit tests with Jest for example component
- React storybook

#### Testing / Reviewing
CI should pass

### CODE REVIEW

File: packages/react/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Getting Started with Create React App
```
Comment (annawen1) [inline packages/react/README.md:1]: Let's just remove the README file for now until we get proper docs written

File: packages/react/.gitignore [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
+# testing
+/coverage
+
+# production
+/build
```
Comment (annawen1) [inline packages/react/.gitignore:12]:
```suggestion
/build
/storybook-static
```

**Review by annawen1 — DISMISSED** [review]
Looks good, some minor clean up

File: packages/react/public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
[the remark this answers is not in the document]
Comment (Gururajj77) [reply packages/react/public/manifest.json:1]: done

File: packages/react/README.md (hunk 1 above)
Comment (Gururajj77) [reply packages/react/README.md:1]: done

**Review by annawen1 — APPROVED** [review]

**Pull request merged**